src/logic/export/RectLabelsExporter.ts

```diff
@@ -54,6 +62,81 @@ export class RectLabelsExporter {
         return labelRectsString.join("\n");
     }
 
+    private static exportAsVOC(): void {
+        let zip = new JSZip();
+        store.getState().editor.imagesData.forEach((imageData: ImageData) => {
+            const fileContent: string = RectLabelsExporter.wrapImageIntoVOC(imageData);
+            if (fileContent) {
+                const fileName : string = imageData.fileData.name.replace(/\.[^/.]+$/, ".xml");
+                zip.file(fileName, fileContent);
+            }
+        });
+
+        const projectName: string = store.getState().editor.projectName
+            .toLowerCase()
+            .replace(' ', '-');
+
+        const date: string = moment().format('YYYYMMDDhhmmss');
+        zip.generateAsync({type:"blob"})
+            .then(function(content) {
+                saveAs(content, "labels_" + projectName + '_' + date + ".zip");
+            });
+    }
+
+    private static wrapRectLabelsIntoVOC(imageData: ImageData): string {
+        if (imageData.labelRects.length === 0 || !imageData.loadStatus)
+            return null;
+
+        const labelNamesList: string[] = store.getState().editor.labelNames;
+        const labelRectsString: string[] = imageData.labelRects.map((labelRect: LabelRect) => {
+            const labelFields = [
+                "\t<object>",
+                "\t\t<name>" + labelNamesList[labelRect.labelIndex] + "</name>",
+                "\t\t<pose>Unspecified</pose>",
+                "\t\t<truncated>Unspecified</truncated>",
+                "\t\t<difficult>Unspecified</difficult>",
+                "\t\t<bndbox>",
+                "\t\t\t<xmin>" + Math.round(labelRect.rect.x) + "</xmin>",
+                "\t\t\t<ymin>" + Math.round(labelRect.rect.y) + "</ymin>",
+                "\t\t\t<xmax>" + Math.round(labelRect.rect.x + labelRect.rect.width) + "</xmax>",
+                "\t\t\t<ymax>" + Math.round(labelRect.rect.y + labelRect.rect.height) + "</ymax>",
+                "\t\t</bndbox>",
+                "\t</object>"
+            ];
+            return labelFields.join("\n")
+        });
+        return labelRectsString.join("\n");
+    }
+
+    private static wrapImageIntoVOC(imageData: ImageData): string {
+        const labels: string = RectLabelsExporter.wrapRectLabelsIntoVOC(imageData);
+        const projectName: string = store.getState().editor.projectName
```

Comment:
już któryś raz xD to ewidentnie do utila, albo niech raz się zbuduje w tym obiekcie który będzie dostarczany adapterom